### PR TITLE
Aggregate a sweep of numeric rows at their own kind

### PR DESCRIPTION
**Pull request by jhuggins**
> **Note.** This replaces #423, which GitHub auto-closed when the branch was
> briefly force-pushed with no commits ahead of its base, and which the API
> refuses to reopen. Same branch, same content.

Stacked on #422 (`dev/sample-draw-level`). Part of #398.

## What changes

A `Function` swept over a batch collected numeric rows into a single-field `NumericRecordBatch` keyed by the function's own name. Both array-row branches of `_make_stack` — the row-wise list path and the mapped `jnp.ndarray` path — now build a `NumericArrayBatch`.

```python
out = my_func(p=design)          # a sweep over 6 rows
out["my_func"]                   # before
out.values                       # now
```

## The batch-row branch covered only one kind

`_make_stack` has a branch for rows that are themselves batches, so a row's own levels end up behind the sweep's rather than being read as event shape. It tested for `RecordBatch` only.

That was harmless until #422, where a scalar law's `sample` began returning a `NumericArrayBatch`. Those rows then fell past the branch into the generic array handling, which read the rows' `draw` axis as event shape and dropped the level — exactly the failure the branch's own comment describes:

> which would take a single-field numeric batch for an array, read its inner batch axis as an event axis, and discard the level names with it

Concretely, `sample(dist_array, sample_shape=(7,))`:

| | before | now |
| --- | --- | --- |
| `batch_shape` | `(4,)` | `(4, 7)` |
| `level_names` | `("dist",)` | `("dist", "draw")` |
| element shape | `(7,)` | `()` |

The record-valued path was already `("dist", "draw")` over `(3, 5)`, so the two agree again. This is what design V.9 states: the outer level ranges over the laws and the inner over each law's draws, "a `RecordBatch` for a record-drawing law, and the batch form of the declared kind otherwise".

## Shared agreement checks

The all-or-nothing and row-agreement checks move into `_agreeing_batch_rows`, used by both kinds. It matches on the batch *family* rather than the exact class, so a `RecordBatch` row and a `NumericRecordBatch` row still stack together — they hold the same thing and only one says so in its name.

## Tests

24 failures, matching the measurement recorded when this stage was scoped. All were the old contract: 12 indexing the aggregate by the function's name, 12 asserting `isinstance(..., NumericRecordBatch)`. No functional gaps.

Fixing the batch-row branch resolved 3 of them before any test was touched.

New coverage in `TestBatchValuedRowAggregation`, mirroring what existed for `RecordBatch` rows:

- array-batch rows stack with the sweep level in front
- array-batch rows disagreeing on their multiplicity are refused
- mixing an array-batch row with a record-batch row is refused

Suite: 4823 passed, 54 skipped. `ruff check` / `ruff format --check` clean.

## Not in scope

`_wrap_declared_function_output` — the path taken when a `Function` carries an `output_template` — still needs the same kind dispatch. That is the next piece.

### CODE REVIEW

**Comment by jhuggins** [comment]
Superseded by #436. The stack grew to eleven PRs; the same commits are regrouped into four, split by concern rather than by the order they were written. No content changed — the four branches carry these commits verbatim (plus one commit recording that a batch of empty records stays refused). Closing to keep review in one place.
